semgrep/semgrep/semgrep_types.py

```diff
@@ -118,9 +119,42 @@ def pattern_names_for_operators(operators: List[Operator]) -> List[str]:
 class Range(NamedTuple):
     start: int
     end: int
+    vars: Mapping[str, Any]
```

Comment:
should this be a property of PatternMatch rather than Range?

Reply:
I need it for the equality and inclusion operations, so I'd have to call it all over the place if I did that. I think it makes more sense for an object to declare its own equality.